### PR TITLE
ref(config): Validate some Kafka configuration on startup

### PR DESCRIPTION
**Pull request by Dav1dde**
Checks for unused topic configurations but does not fail, this is useful when preparing to rollout a new topic or rolling back to an older Relay version.

Validates that configuration for every topic can be built.

related to https://github.com/getsentry/relay/issues/3404

#skip-changelog

### CODE REVIEW

**Review by olksdr — COMMENTED** [review]
Is it possible to add the test for this? Maybe integration test?

File: relay-kafka/src/config.rs [hunk 1]
```diff
@@ -82,18 +82,20 @@ impl KafkaTopic {
...
             $(
                 #[serde(alias = $default_topic)]
-                #[doc=$doc]
+                #[doc = $doc]
                 pub $field_name: TopicAssignment,
             )*
+
+            /// Additional topic assignments configured but currently unused by this Relay instance.
+            #[serde(flatten)]
+            pub unused: BTreeMap<String, TopicAssignment>,
```
Comment (jjbayer) [inline relay-kafka/src/config.rs:98]: Should we deny unknown fields instead? https://serde.rs/container-attrs.html#deny_unknown_fields

Comment (Dav1dde) [reply relay-kafka/src/config.rs:98]: We could, see the PR description, deny unknown adds some behaviour we maybe don't want.

Comment (jjbayer) [reply relay-kafka/src/config.rs:98]: Thanks, I should have read the description. So the error is more like a warning, maybe we should start reporting warnings to sentry.

We could theoretically log the error in a `deserialize_with` function on `unused` (not sure if that's better, just brainstorming here).

**Review by jjbayer — APPROVED** [review]

**Review by iker-barriocanal — APPROVED** [review]
A test wuld be great

File: relay-kafka/src/config.rs (hunk 1 above)
Comment (Dav1dde) [reply relay-kafka/src/config.rs:98]: Yeah, initially I planned to do a warning, but I really would like it to be reported to Sentry!

> We could theoretically log the error in a deserialize_with function on unused (not sure if that's better, just brainstorming here).

Open to changing it, but with the current solution the thing I don't like is the `unused_topics` accessor. We'd get rid of that with that.

Maybe we should move the validation to `Config` and then we can just call `config.validate()`, potentially even having a `ValidatedConfig(Config)`.

**Pull request merged**